utils/containers/lock_free/lf_mpmc_queue.h.tpl

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __UTILS_CONTAINERS_LOCK_FREE_LF_MPMC_QUEUE_{TYPE}_H__
+#define __UTILS_CONTAINERS_LOCK_FREE_LF_MPMC_QUEUE_{TYPE}_H__
+
+#include <ck_fifo.h>
+
+#include "common/errors.h"
+
+#define LF_MPMC_QUEUE_IS_EMPTY(f) CK_FIFO_MPMC_ISEMPTY(f)
+#define LF_MPMC_QUEUE_FIRST(f) CK_FIFO_MPMC_FIRST(f)
+#define LF_MPMC_QUEUE_NEXT(f) CK_FIFO_MPMC_NEXT(m)
+#define LF_MPMC_QUEUE_FOREACH(fifo, entry) CK_FIFO_MPMC_FOREACH(fifo, entry)
+#define LF_MPMC_QUEUE_FOREACH_SAFE(fifo, entry, T) \
+  CK_FIFO_MPMC_FOREACH_SAFE(fifo, entry, T)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// A generic concurrent lock-free Multi Producer Multi Consumer queue
+typedef ck_fifo_mpmc_t lf_mpmc_queue_{TYPE}_t;
```

Comment:
seems like we could do:
`typedef ck_fifo_mpmc_t lf_mpmc_queue_t;` 
this does not provide type safety, so we might as well get rid of the {TYPE} param?

Reply:
I do agree that it doesn't provide type safety as they would all come down to the same type `ck_fifo_mpmc_t`. But it does show intent and clarity when declaring a queue.